src/gt4py/next/embedded/nd_array_field.py

```diff
@@ -428,6 +434,24 @@ def _slice(
         assert common.is_relative_index_sequence(slice_)
         return new_domain, slice_
 
+    def __copy__(self) -> NdArrayField:
+        # Note: `copy` copies the data, following NumPy behavior
+        allocation_ns = (
+            self._allocation_ns
+            if self._allocation_ns is not None
+            else _allocators.get_array_allocation_namespace(self.array_ns)
+        )
+        ndarray_copy = allocation_ns.asarray(
+            self.ndarray,
+            domain=self.domain,
+            dtype=self.dtype,
+            copy=True,  # aligned_index???
```

Comment:
Actually this is a missing piece in the current implementation. We should probably provide this as a default in all functions.